rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,510 @@
+                    )
+
+                    serves_kitchen_space = (
+                        True
+                        if ventilation_space_type_p
+                        == LIGHTING_SPACE.FOOD_AND_BEVERAGE_SERVICE_KITCHEN_COOKING
+                        else False
+                    )
+                else:
+                    all_ventilation_space_types_defined = False
+
+            hvac_sys_and_assoc_zones_largest_exhaust_source[
+                hvac_id_b
+            ] = get_hvac_sys_and_assoc_zones_largest_exhaust_source(rmi_b, hvac_id_b)
+
+        for hvac_id_p in get_list_hvac_systems_associated_with_zone(rmi_p, zone_id_b):
+            hvac_p = find_one(
+                f"$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[?(@.id == {hvac_id_p})]",
+                rmi_p,
+            )
+            ER_modeled_in_proposed = (
+                True
+                if (
+                    getattr_(hvac_p, "HVAC", "fan_system").get("air_energy_recovery")
+                    is not None
+                    and getattr_(
+                        hvac_p,
+                        "HVAC",
+                        "fan_system",
+                        "air_energy_recovery",
+                        "energy_recovery_type",
+                    )
+                    != ENERGY_RECOVERY.NONE
+                )
+                else False
+            )
+
+            serves_zones_that_have_dehumid_heat_recovery = (
+                True
+                if (
+                    hvac_p.get("cooling_system") is not None
+                    and hvac_p["cooling_system"].get("dehumidification_type")
+                    == DEHUMIDIFICATION.SERIES_HEAT_RECOVERY
+                )
+                else False
+            )
+
+        sys_type_heating_only = (
+            True
+            if any(
+                [
+                    baseline_system_type_compare(
+                        system_type, applicable_sys_type, False
+                    )
+                    for system_type in get_baseline_system_types(rmi_b).keys()
+                    for applicable_sys_type in APPLICABLE_SYS_TYPES
+                ]
+            )
+            else False
+        )
+
+        max_thermostat_heating_setpoint_schedule_p = max(
+            [
+                max(thermostat_schedule)
+                for thermostat_schedule_id in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].thermostat_heating_setpoint_schedule",
+                    rmi_p,
+                )
+                for thermostat_schedule in find_one(
+                    f"$.schedules[?(@.id == {thermostat_schedule_id})].hourly_values",
+                    rmi_p,
+                )
+            ]
+        )
+        serves_zones_heated_to_60_or_higher_in_proposed = (
+            True
+            if max_thermostat_heating_setpoint_schedule_p > 60 * ureg("F")
+            else False
+        )
```

Comment:
This shall be calculated in the other loop (the one above), not in here.
Note in the RDS, this flag is inside the zone loop but outside the `for hvac_p in hvac_sys_list: `

Reply:
addressed.